test/search_index_ddoc_update_tests.erl

```diff
@@ -0,0 +1,277 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(search_index_ddoc_update_tests).
+
+-include_lib("couch/include/couch_eunit.hrl").
+-include_lib("couch/include/couch_db.hrl").
+-include("dreyfus.hrl").
+
+-define(DDOC, {[
+    {<<"_id">>, <<"_design/searchddoc">>},
+    {<<"indexes">>, {[
+        {<<"other">>, {[
+            {<<"index">>, <<"function(doc){if(doc.blah){index(\\\"blah\\\", doc.blah);}}">>}
+        ]}},
+        {<<"default">>, {[
+            {<<"index">>, <<"function(doc){if(doc.blah){index(\\\"default\\\", doc.blah);}}">>}
+        ]}}
+    ]}}
+]}).
+
+-define(USER, "admin").
+-define(PASS, "pass").
+-define(AUTH, {basic_auth, {?USER, ?PASS}}).
+
+
+start() ->
+    Ctx = test_util:start_couch([chttpd, dreyfus]),
+    ok = config:set("admins", ?USER, ?PASS, _Persist=false),
+    Ctx.
+
+setup(PortType) ->
```

Comment:
Here is one example of it https://github.com/apache/couchdb/blob/master/src/couch/test/couchdb_compaction_daemon_tests.erl#L62